### DISCUSSION TITLE
Not able to apply the patches from #1521 PR

### DISCUSSION THREAD

**Discussion by nrpt-m** [gitpython-developers/GitPython#1529] [Q&A] [answered]
Hi @Byron ,

I am working on fixing the CVE-2022-24439 on our source code. Our product is currently having python3-git with 3.1.27 version. As I got to know that #1521 PR is fixing the CVE-2022-24439 so, started backporting the patches. But, after applying patches observed that in our python3-git source code there is no "test/" directory available and due to which patches are failing to apply.

Later on found that there no test/ directory in original tarball downloaded from this https://files.pythonhosted.org/packages/source/G/GitPython/GitPython-3.1.27.tar.gz 

Could you please tell me if I can ignore the changes in test/ directory & apply the patches then, will it fix the CVE-2022-24439 issue ? If not then how to fix this CVE-2022-24439 in this source code where test/ directory is not at available ?

Thanks.

**Comment by stsewd** [comment, marked as answer]
You should be good ignoring the changes to tests, make also sure to apply the changes from https://github.com/gitpython-developers/GitPython/pull/1518.

**Reply by nrpt-m** [reply]
Hi @stsewd ,

Have generated 2 different patches from PR 1518 & 1521 which includes all the changes except test/ directory changes.
Have attached both the patches here. Could you please have a quick review & confirm if these patches will be able to fix the CVE-2022-24439 ?
[0001-python3-git-CVE-2022-24439-fix-from-PR-1518.patch](https://github.com/gitpython-developers/GitPython/files/10368965/0001-python3-git-CVE-2022-24439-fix-from-PR-1518.patch)
[0001-python3-git-CVE-2022-24439-fix-from-PR-1521.patch](https://github.com/gitpython-developers/GitPython/files/10368966/0001-python3-git-CVE-2022-24439-fix-from-PR-1521.patch)

Thanks.

**Reply by Byron** [reply]
Great to hear, maybe others might find these useful.

Please don't wait for me, I won't be reviewing them.

**Reply by nrpt-m** [reply]
@stsewd Could you please check these patches & confirm if these patches will be able to fix the CVE-2022-24439 ?

**Reply by stsewd** [reply]
Hi there, I've been busy, took a quick look and patches look okay.

**Reply by nrpt-m** [reply]
Thanks a lot @stsewd & @Byron !!

**Reply by philsuth** [reply]
@nrpt-m  - In line 294 of your PR1521 patch I think there's an indentation error (breaking line 700 of remote.py): the "url = Git.polish_url(url)"  is indented one space too far.  Noticed this over in yocto/poky where the patches have been recently merged into Kirkstone. Will you follow up there or should I?

**Reply by nrpt-m** [reply]
@philsuth, Thanks for your sharp observations. It would be great help if you could follow up there.
